server/src/main/java/org/opensearch/indices/replication/PrimaryShardReplicationSource.java

```diff
@@ -78,14 +80,33 @@ public void getSegmentFiles(
     ) {
         final Writeable.Reader<GetSegmentFilesResponse> reader = GetSegmentFilesResponse::new;
         final ActionListener<GetSegmentFilesResponse> responseListener = ActionListener.map(listener, r -> r);
+        // Few of the below assumptions and calculations are added for experimental release of segment replication feature in 2.3
+        // version.These will be changed in next release.
+
+        // Storing the size of files to fetch in bytes.
+        long sizeOfSegmentFiles = 0;
+        for (int i = 0; i < filesToFetch.size(); i++) {
+            sizeOfSegmentFiles += filesToFetch.get(i).length();
+        }
+        // Making sure files size is in correct format to perform time calculation.
+        sizeOfSegmentFiles = Math.abs(sizeOfSegmentFiles);
+        // Maximum size of files to fetch (segment files), that can be processed in 1 minute for a m5.xlarge machine.
+        long baseSegmentFilesSize = 300000000;
+
+        long timeToGetSegmentFiles = 1;
+        // Formula for calculating time needed to process a replication event's files to fetch process
+        timeToGetSegmentFiles += sizeOfSegmentFiles / baseSegmentFilesSize;
```

Comment:
Can this be simplified with below ?
```
final long timeToGetSegmentFiles = Math.max (1, sizeOfSegmentFiles / baseSegmentFilesSize);
```

Reply:
Let's take example where baseline is 100000000 bytes (100mb) and size of segment files is 250000000(250mb). When we do a sizeOfSegmentFiles / baseSegmentFilesSize result will be 2min. But we need 3min to process this size of segment, so we will have a timeout. To avoid this we need to add a 1 before and after " sizeOfSegmentFiles / baseSegmentFilesSize"